README.md

```diff
@@ -58,81 +54,52 @@ Include this badge in your readme if you make a new module that implements inter
 
-### Local resolver methods
 
-#### `resolver.resolve(binaryBlob, path, callback)`
+### `cid(binaryBlob, [options])`
 
-> resolves a path in block, returns the value and or a link and the partial missing path. This way the IPLD Resolver can fetch the link and continue to resolve.
+> Return the CID of the binary blob.
 
-`callback` must have the signature `function (err, result)`, where `err` is an Error if the function fails and `result` is an object with the following keys:
+Possible `options` are:
+  - `version` (`number`, default: 1): the CID version to be used
+  - `hashAlg` (`Multicodec`, default: the one the format specifies): the hash algorithm to be used
 
-- value: <> - The value resolved or an IPLD link if it was unable to resolve it through.
-- remainderPath: <> - The remaining path that was not resolved under block scope.
+This can be used to verify that some data actually has a certain CID.
 
-If `path` is the root `/`, the result is a nested object that contains all paths that `tree()` returns. The values are the same as accessing them directly with the full path. Example:
-
-`tree()` returns:
-
-```JSON
-["author/name", "author/email"]
-```
-
-`resolve(binaryblob, "/", callback)` would then have as a result:
-
-```JSON
-{
-  "author": {
-    "name": "vmx",
-    "email": "vmx@example.com"
-  }
-}
-```
-
-Numbers within a path are interpreted as an array.
-
-#### `resolver.tree(binaryBlob, callback)`
-
-> returns all the paths available in this block.
-
-`callback` must have the signature `function (err, result)`, where `err` is an Error if the function fails and `result` is a list of path such as `["/foo", "/bar", "/author/name", ...]`.
 
 ### Properties
 
 #### `defaultHashAlg`
 
-> Default hash algorithm of the format
+> Default hash algorithm of the format,
+
+Most formats have one specific hash algorithm, e.g. Bitcoin’s is `dbl-sha2-256`. CBOR can be used with any hash algorithm, though the default in the IPFS world is `sha256`. `defaultHashAlg` is used in the `cid()` call if no hash algorithm is given. The value of `defaultHashAlg` is of type `Multicodec` must be one code defined in the [Multihash Table](https://github.com/multiformats/multihash#table-for-multihash).
 
-Most formats have one specific hash algorithm, e.g. Bitcoin’s is `dbl-sha2-256`. CBOR can be used with any hash algorithm, though the default in the IPFS world is `sha256`. `defaultHashAlg` is used in the `util.cid()` call if no hash algorithm was given. The value of `defaultHashAlg` must be one defined in the [Multihash Table](https://github.com/multiformats/multihash#table-for-multihash-v100-rc-semver).
+#### `format`
 
-#### `multicodec`
+> Identifier for the format implementation.
 
-> Identifier for the format implementation
+The `format` property of type `Multicodec` is used to register a format implementation in IPLD. It needs to be one of the codes specified in the [Multicodec Table](https://github.com/multiformats/multicodec#multicodec-table).
```

Comment:
If we are moving to using the hex value rather than the string value do we actually need to require in *this* API that the code is already registered in the table?

At some point this table is going to get much harder to add things to and it would be nice to open up an easier path for developers who are working on new codecs.

Reply:
To be clear, I'm recommending that we change from "It needs to" to "It *should* be".

Reply:
You're absolutely right. This is what I wanted to say, it's just a language thing. I'll try to formulate it in a clearer way.